pool.go

```diff
@@ -116,10 +116,25 @@ func (p *Pool) ticktock() {
 	defer ticker.Stop()
 
 	for range ticker.C {
+		if p.IsClosed() {
```

Comment:
I think the better way to do this is to use context and `select` in `for` loop, like `purgeStaleWorkers()` does.
So you should pass the context to `ticktock()` and use an atomic int32 to indicate whether `ticktock()` is done, handle these properly in `ReleaseTimeout()` and `Reboot()`.

In short, just do what exactly `purgeStaleWorkers()` has already done.

Reply:
Addressed, but I'm still concerned if atomic int32 is helpful for ticktock case since there is not work-related awaiting logic. lmk if you need anything else